saleor/graphql/account/bulk_mutations/customer_bulk_update.py

```diff
@@ -0,0 +1,586 @@
+                )
+                continue
+
+            customer_id = cleaned_input.get("id")
+            external_ref = cleaned_input.get("external_reference")
+            data = cleaned_input["input"]
+            shipping_address_input = data.pop(SHIPPING_ADDRESS_FIELD, None)
+            billing_address_input = data.pop(BILLING_ADDRESS_FIELD, None)
+
+            filter_customer = list(
+                filter(
+                    cls._get_customer(customer_id, external_ref),
+                    customers_list,
+                )
+            )
+            if filter_customer:
+                try:
+                    shipping_address = None
+                    billing_address = None
+
+                    old_instance = filter_customer[0]
+                    new_instance = cls.construct_instance(copy(old_instance), data)
+                    cls.clean_instance(info, new_instance)
+
+                    if shipping_address_input:
+                        shipping_address = cls.update_address(
+                            info,
+                            new_instance,
+                            shipping_address_input,
+                            SHIPPING_ADDRESS_FIELD,
+                        )
+                    if billing_address_input:
+                        billing_address = cls.update_address(
+                            info,
+                            new_instance,
+                            billing_address_input,
+                            BILLING_ADDRESS_FIELD,
+                        )
+
+                    instances_data_and_errors_list.append(
+                        {
+                            "instance": new_instance,
+                            "old_instance": old_instance,
+                            "errors": index_error_map[index],
+                            SHIPPING_ADDRESS_FIELD: shipping_address,
+                            BILLING_ADDRESS_FIELD: billing_address,
+                        }
+                    )
+                except ValidationError as exc:
+                    cls.format_errors(index, exc, index_error_map)
+                    instances_data_and_errors_list.append(
+                        {"instance": None, "errors": index_error_map[index]}
+                    )
+            else:
+                index_error_map[index].append(
+                    CustomerBulkUpdateError(
+                        message="Customer was not found.",
+                        code=CustomerBulkUpdateErrorCode.NOT_FOUND.value,
+                    )
+                )
+                instances_data_and_errors_list.append(
+                    {"instance": None, "errors": index_error_map[index]}
+                )
+
+        return instances_data_and_errors_list
+
+    @classmethod
+    def save_customers(cls, instances_data_with_errors_list, manager):
+        old_instances = []
+        customers_to_update = []
+        address_to_create = []
```

Comment:
Naming can be more consistent, some lists use singular and some plural forms. This should be: `addresses_to_create`